controllers/ocsinitialization/ocsinitialization_controller.go

```diff
@@ -252,14 +256,72 @@ func (r *OCSInitializationReconciler) ensureRookCephOperatorConfigExists(initial
 			Namespace: initialData.Namespace,
 		},
 	}
-	err := r.Client.Create(r.ctx, rookCephOperatorConfig)
-	if err != nil && !errors.IsAlreadyExists(err) {
-		r.Log.Error(err, fmt.Sprintf("Failed to create %s configmap", util.RookCephOperatorConfigName))
+
+	opResult, err := ctrl.CreateOrUpdate(r.ctx, r.Client, rookCephOperatorConfig, func() error {
+
+		csiPluginDefaults := defaults.DaemonPlacements[defaults.CsiPluginKey]
+		csiProvisionerDefaults := defaults.DaemonPlacements[defaults.CsiProvisionerKey]
+
+		csiPluginTolerations := r.getCsiTolerations(defaults.CsiPluginKey)
+		csiProvisionerTolerations := r.getCsiTolerations(defaults.CsiProvisionerKey)
+
+		if rookCephOperatorConfig.Data == nil {
+			rookCephOperatorConfig.Data = make(map[string]string)
+		}
+
+		updateTolerationsConfigFunc := func(tolerations, defaults []corev1.Toleration, configMapKey string, modifiedFlag *bool) error {
+
+			if tolerations != nil {
+				updatedTolerations := append(tolerations, defaults...)
+				tolerationsYAML, err := yaml.Marshal(updatedTolerations)
+				if err != nil {
+					return err
+				}
+				rookCephOperatorConfig.Data[configMapKey] = string(tolerationsYAML)
+				*modifiedFlag = true
+			} else if tolerations == nil && *modifiedFlag {
+				delete(rookCephOperatorConfig.Data, configMapKey)
+				*modifiedFlag = false
+			}
+
+			return nil
+		}
```

Comment:
Why do we need to define a new function for each reconcile iteration? What prevents us from having this as a package top-level functions?

Reply:
The func will become bit complicated if we define it in the top level as we are accessing the other variables as well defined locally. We will have to pass them. 

Also does it really matter? Because it has a unique use case. It can not be reused by some one else.